nifi-extension-bundles/nifi-network-bundle/nifi-network-processors/src/main/java/org/apache/nifi/processors/network/pcap/SplitPCAP.java

```diff
@@ -0,0 +1,235 @@
+                    int currentPcapTotalLength = PCAP.PCAP_HEADER_LENGTH;
+
+                    final PCAP templatePcap = new PCAP(new ByteBufferInterface(pcapHeader));
+
+                    while (bInStream.available() > 0) {
+
+                        byte[] packetHeader = new byte[Packet.PACKET_HEADER_LENGTH];
+                        bInStream.read(packetHeader, 0, Packet.PACKET_HEADER_LENGTH);
+                        Packet currentPacket = new Packet(packetHeader, templatePcap);
+
+                        if (currentPacket.totalLength() > pcapMaxSize) {
+                            throw new ProcessException("PCAP contains a packet larger than the max size.");
+                        }
+
+                        if (currentPacket.isInvalid()) {
+                            throw new ProcessException("PCAP contains an invalid packet.");
+                        }
+
+                        byte[] packetbody = new byte[(int) currentPacket.expectedLength()];
+
+                        bInStream.read(packetbody, 0, (int) currentPacket.expectedLength());
+                        currentPacket.setBody(packetbody);
+
+                        if (currentPcapTotalLength + currentPacket.totalLength() > pcapMaxSize) {
+
+                            templatePcap.packets().addAll(loadedPackets);
+                            FlowFile newFlowFile = session.create(flowFile);
+                            try (final OutputStream out = session.write(newFlowFile)) {
+                                out.write(templatePcap.readBytesFull());
+                                splitFiles.add(newFlowFile);
+                            }
+
+                            loadedPackets.clear();
+                            loadedPackets.add(currentPacket);
+                            currentPcapTotalLength = PCAP.PCAP_HEADER_LENGTH;
+                            templatePcap.packets().clear();
+                        }
+
+                        loadedPackets.add(currentPacket);
+                        currentPcapTotalLength += currentPacket.totalLength();
+                    }
+
+                    // If there are any packets left over, create a new flowfile.
+                    if (!loadedPackets.isEmpty()) {
+                        templatePcap.packets().addAll(loadedPackets);
+                        FlowFile newFlowFile = session.create(flowFile);
+                        try (final OutputStream out = session.write(newFlowFile)) {
+                            out.write(templatePcap.readBytesFull());
+                            splitFiles.add(newFlowFile);
+                        }
+                    }
+                }
+            });
+        } catch (ProcessException e) {
+            session.remove(splitFiles);
+            splitFiles.clear();
+            session.putAttribute(flowFile, ERROR_REASON_LABEL, e.getMessage());
+            session.transfer(flowFile, REL_FAILURE);
```

Comment:
A log statement should be added. 

```suggestion
            getLogger().error("Failed to split {}", flowFile, e);
            session.remove(splitFiles);
            splitFiles.clear();
            session.putAttribute(flowFile, ERROR_REASON_LABEL, e.getMessage());
            session.transfer(flowFile, REL_FAILURE);
```